calculator/main.py

```diff
@@ -26,7 +27,22 @@ def perform_operation(calc, operation, params):
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
     """
-    pass
+   
+   
+    if not type(params) is tuple: 
+        raise InvalidParams('Given params are invalid.')
+    elif not params:
```

Comment:
This is redundant, if params were not passed the function call would fail, and if they were `None` your check to see if they were a tuple would fail.